### PR TITLE
Nixpkgs upgrade

### LINKED ISSUE #9500: Upgrade nixpkgs to include Go 1.13
I was requested by @cammellos to upgrade the current Go version we are using:
https://github.com/status-im/status-react/blob/5ecc7590d98fafd834a065dad82c01649b47c973/nix/derivation.nix#L20
But in order to do that I might have to do a full upgrade of our [`nixpkgs`](https://github.com/status-im/nixpkgs) fork.

Previous Go upgrade for reference: https://github.com/status-im/nixpkgs/commit/35e0ad9f6d83f431fe5a156793e44f095b8039ab

### PR DESCRIPTION
**Pull request by jakubgs**
This upgrades a LOT of the tools and libraries that Nix provides by upgrading our fork of `nixpkgs`.

Changes:
* Upgrades `nixpkgs`
* Upgrades Go: `1.12` > `1.13`
* Upgrades Android SDK: `28.0.3` > `29.0.2`
* Makes `uploadToSauceLabs()` not use a pure Nix shell to avoid issues with certs
* Narrows down target for `scripts/copy-translations.sh` to `targets.mobile.jsbundle`
* Removes Desktop builds from CI since work on them is on-hold and new `nixpkgs` break them
* Fixes an issue with our patched `mkShell` in `shell.nix` by inheriting instead of setting `shellHook`
* Fixes gomobile status-go build by setting `GO111MODULE=off`
* Fixes gradle accepting empty `STATUS_GO_SRC_OVERRIDE` value

Resolves: #9500

### CODE REVIEW

File: nix/status-go/build-status-go.nix [hunk 1]
```diff
@@ -53,11 +53,8 @@ let
     '';
 
     # replace hardcoded paths to go package in /nix/store, otherwise Nix will fail the build
-    preFixup = ''
-      ${preFixup}
-
+    preFixup = preFixup + ''
       find $out -type f -exec ${removeExpr removeReferences} '{}' + || true
-      return
```
Comment (pedropombeiro) [inline nix/status-go/build-status-go.nix:60]: Why did you remove this `return` here? It is required, otherwise the host `preFixup` script will continue executing, which we don't want. Maybe we can add a comment to make it explicit.

File: nix/derivation.nix [hunk 2]
```diff
@@ -60,9 +62,9 @@ in {
 
   shell = {
     buildInputs = unique ([
+      yarn
```
Comment (pedropombeiro) [inline nix/derivation.nix:65]: Why did yarn need to come first in the list?

File: nix/mobile/android/android-env.nix [hunk 3]
```diff
@@ -10,18 +10,18 @@ let
 
   androidComposition = androidenv.composeAndroidPackages {
     toolsVersion = "26.1.1";
-    platformToolsVersion = "28.0.2";
-    buildToolsVersions = [ "28.0.3" ];
+    platformToolsVersion = "29.0.5";
+    buildToolsVersions = [ "29.0.2" ];
```
Comment (pedropombeiro) [inline nix/mobile/android/android-env.nix:14]: I wonder if we should try to read these values from `android/gradle.properties`

File: nix/status-go/build-mobile-status-go.nix [hunk 4]
```diff
@@ -31,10 +31,11 @@ let
       in with targetConfig; ''
       mkdir ${NIX_GOWORKDIR}
 
-      GOPATH=${gomobile.dev}:$GOPATH \
-      PATH=${makeBinPath [ gomobile.bin ]}:$PATH \
-      NIX_GOWORKDIR=${NIX_GOWORKDIR} \
-      ${concatStringsSep " " envVars} \
+      export GO111MODULE=off
```
Comment (pedropombeiro) [inline nix/status-go/build-mobile-status-go.nix:34]: Was there an error that you saw that caused us to need to export the variables?

**Review by pedropombeiro — APPROVED** [review]
Great job! Left a couple of comments.

File: nix/derivation.nix (hunk 2 above)
Comment (jakubgs) [reply nix/derivation.nix:65]: OCD

File: nix/status-go/build-mobile-status-go.nix (hunk 4 above)
Comment (jakubgs) [reply nix/status-go/build-mobile-status-go.nix:34]:
```
Building mobile library for android

type-checking package "github.com/status-im/status-go/mobile" failed (/build/go/src/github.com/status-im/status-go/mobile/multiaccount.go:4:2: could not import encoding/json (go/build: importGo encoding/json: exit status 2
go: cannot find GOROOT directory: /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-go-1.13.4/share/go
```

**Review by vitvly — APPROVED** [review]

File: nix/mobile/android/android-env.nix (hunk 3 above)
Comment (jakubgs) [reply nix/mobile/android/android-env.nix:14]: Not sure how difficult that would be in Nix.

Comment (jakubgs) [reply nix/mobile/android/android-env.nix:14]: I'll do it in another PR, since this fixes iOS fastlane build errors and I want to get this merged.

**Review by flexsurfer — APPROVED** [review]
